checkstyle/src/main.rs

```diff
@@ -0,0 +1,51 @@
+use clap::Parser;
+use std::path::PathBuf;
+use std::process;
+mod separate;
+use separate::separate_regex_matching_files;
+
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    #[arg(short, long)]
+    target_path: String,
+
+    #[arg(short, long)]
+    regex_file_path: String,
+
+    #[arg(short, long)]
+    include: Option<Vec<String>>,
+    #[arg(short, long)]
+    exclude: Option<Vec<String>>,
+}
+
+fn main() -> Result<(), std::io::Error> {
+    let args = Args::parse();
+
+    let regex_file_path = PathBuf::from(&args.regex_file_path);
+    let target_path = PathBuf::from(&args.target_path);
+
+    let include_patterns = args.include;
+    let exclude_patterns = args.exclude;
+
+    let nonmatching_files = separate_regex_matching_files(
+        &regex_file_path,
+        &target_path,
+        include_patterns,
+        exclude_patterns,
+    );
```

Comment:
A single function call is nice, but let's try and make this more composable. I think we want two distinct processes from the highest level here: 
* getting all of the files we're going to consider given the patterns
* filtering out the non-matching ones

We're going to want to have additional steps after this point as well, so let's build our program to allow for that extension to happen easily. Our Rust modules themselves should reflect this as well - let's have one for the regex matching functions and one for the glob matching functions.